kvclientpool.go

```diff
@@ -505,24 +504,24 @@ func (p *kvClientPool) Reconfigure(config *KvClientPoolConfig, cb func(error)) e
 
 		// reconfiguring is successful up until this point, so it can stay in the
 		// current list of clients.  it may be moved later by the Reconfigure callback.
-		numReconfigured++
+
+		// if the pool name changed as part of this reconfigure, move it.
+		if oldPoolName != newPoolName {
+			p.connCountMetric.Record(context.Background(), -1, metric.WithAttributes(
```

Comment:
This is now in the loop so is going to record the metrics for every client that gets reconfigured, is that correct?

Reply:
Yea, but we only record -1/+1 for each client that is reconfigured rather than doing them all with multiple-counts like it had previously.